Projects/05matchGame/index.js

```diff
@@ -0,0 +1,402 @@
+          let cloneSavedCards = [...mergeSavedCards];
+
+          allFaceCards = mergeSavedCards.concat(cloneSavedCards);
+          allFaceCards
+            .map((cardFace) => {
+              allSavedCards.push(cardFace.images.png);
+
+              //randomize the cards from API;
+              return allSavedCards.sort(() => Math.random() - 0.5);
+            })
+            .join("");
+        })
+        .catch(() => {
+          notResponseApi();
+        });
+    }
+
+    //show popup with sysMsg - not response from API;
+    function notResponseApi() {
+      document.querySelector(".pageGame").innerHTML = "";
+      uiElements.$popupContainer.style.display = "block";
+      sysMsg.popupWinLoseNotResponse.innerHTML = sysMsg.notResponseAPI;
+      sysMsg.popupWinLoseNotResponse.classList.add("centerSysMsg");
+    }
+
+    //if "x" is clicked, hide the popup;
+    function closePopup() {
+      document.querySelector(".close").onclick = () => {
+        uiElements.$popupContainer.style.display = "none";
+      };
+    }
+
+    //generate the number of card slots;
+    async function handleStartingGame() {
+      uiElements.$containerCards.innerHTML = "";
+
+      generateCardSlots();
+      flipOnClick();
+      generateTable();
+      uiElements.$resetBtn.addEventListener("click", resetBtn);
+    }
+
+    //function to generate the slots with backCard and faceCard(from API);
+    function generateCardSlots() {
+      for (let i = 0; i < slots; i++) {
+        HTMLCards = `<div class="cardSlot" >
+        <div class="flipCard">
+            <div class="backCard">
+                <img src="images/cardback_158.png" alt="backCard"/>
+            </div>
+            <div class="faceCard">
+              <img src="${allSavedCards[i]}"/>
+            </div>
+        </div>`;
+
+        uiElements.$containerCards.innerHTML += HTMLCards;
+      }
+    }
+
+    //function to add and remove class on click;
+    function flipOnClick() {
+      let flipCards = document.querySelectorAll(".flipCard");
+      for (let i = 0; i < flipCards.length; i++) {
+        matchFlipCards = function (event) {
+          // make target = parent element (in this case parent = (".flipcard"))
+          let clickedCard = event.currentTarget;
+
+          //if clickedCard is different from the first clicked card - add class and unlock the card;
+          if (clickedCard !== cardOne && !lockCards) {
+            clickedCard.classList.add("rotateOnClick");
+
+            //if first clicked card = false; first clicked card will be the card user click;
+            if (!cardOne) {
+              return (cardOne = clickedCard);
+            }
+            //second clicked card = next clicked card;
+            cardTwo = clickedCard;
+            lockCards = true;
+            //select first clicked card img and next one clicked card img;
+            let cardOneImg = cardOne.querySelector(".faceCard img").src;
```

Comment:
simplicity - faceCard please rename this class to 'face', it's clear enough because you say that
CardOne -> query selector -> .face -> img -> src

Reply:
@kvmladenov done